tika-parsers/src/main/java/org/apache/tika/parser/recognition/dl4j/DL4JImageRecogniser.java

```diff
@@ -0,0 +1,165 @@
+       private static final LineConsumer IGNORED_LINE_LOGGER = new LineConsumer() {
+        @Override
+        public void consume(String line) {
+            LOG.debug(line);
+        }
+    };
+	 private static final String HOME_DIR = System.getProperty("user.home");
+    private static final String BASE_DIR = ".dl4j/trainedmodels";
+    private static String MODEL_DIR=HOME_DIR+File.separator+BASE_DIR;
+    private static String MODEL_DIR_PREPROCESSED=MODEL_DIR+File.separator+"tikaPreprocessed"+File.separator;
+    @Field private String modelType="VGG16";
+    @Field private File modelFile;
+    @Field private String outPattern = "(.*) \\(score = ([0-9]+\\.[0-9]+)\\)$";
+	File locationToSave ;
+    private boolean available = false;
+    ComputationGraph model;
+    public Set<MediaType> getSupportedMimes() {
+        return SUPPORTED_MIMES;
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return available;
+    }
+
+    @Override
+    public void initialize(Map<String, Param> params) throws TikaConfigException {
+        try {
+        	if(modelType.equals("VGG16NOTOP"))
+            {
+        		throw new TikaConfigException("VGG16NOTOP is not supported right now");
+				/*# TODO hookup VGGNOTOP by uncommenting following code once the issue is resolved by dl4j team
+				modelFile = new File(MODEL_DIR_PREPROCESSED+File.separator+"vgg16_notop.zip");
+				locationToSave= new File(MODEL_DIR+File.separator+"tikaPreprocessed"+File.separator+"vgg16.zip");*/
+            }else if(modelType.equals("VGG16"))
+            {
+            	modelFile = new File(MODEL_DIR_PREPROCESSED+File.separator+"vgg16.zip");
+				locationToSave= new File(MODEL_DIR+File.separator+"tikaPreprocessed"+File.separator+"vgg16.zip");
+            }
+        	if(!modelFile.exists())
+    		{
+                    modelFile.getParentFile().mkdirs();
+                    LOG.warn("Preprocessed Model doesn't exist at {}", modelFile);
+                    TrainedModelHelper helper;
+                   if(modelType.equals("VGG16NOTOP"))
+                    {
+                    	helper = new TrainedModelHelper(TrainedModels.VGG16NOTOP);
+                    }else if(modelType.equals("VGG16"))
+                    {
+                    	helper = new TrainedModelHelper(TrainedModels.VGG16);
+                    }else
+                    {
+                    	throw new TikaConfigException("Unknown or unsupported model");
+                    }
+                    model = helper.loadModel();
+    				/*model= KerasModelImport.importKerasModelAndWeights(MODEL_DIR+File.separator+"VGG16NoTop.json",MODEL_DIR+File.separator+"vgg16_weights_th_dim_ordering_th_kernels_notop",false);*/
+    				LOG.info("Saving the Loaded model for future use. Saved models are more optimised to consume less resource.");
+    				ModelSerializer.writeModel(model,locationToSave,true);
+    				available=true;
+    		}else
```

Comment:
Please fix the code format issues, specifically these:
+ missing space around the keywords
+ `{` that take unnecessary extra lines:
   prefer 
```
if (...) {
}
```
instead of 
```
if (...) 
{
}
```
+ dead code / useless commented code that are not documentation
<img width="1279" alt="screen shot 2017-03-31 at 10 22 40 am" src="https://cloud.githubusercontent.com/assets/1865964/24554556/0a0271dc-15fc-11e7-9dfa-ca49f9224d77.png">


I strongly recommend reading and remembering Java Code Conventions from Sun/Oracle which is here http://www.oracle.com/technetwork/java/codeconventions-150003.pdf
and then configure your IDE to do it automatically for you.